examples/mongodb/src/db.rs

```diff
@@ -0,0 +1,92 @@
+use rocket_contrib::databases::mongodb::db::{Database,ThreadedDatabase};
+use rocket_contrib::databases::mongodb::{bson,doc,oid::{ObjectId}};
+
+use models::Todo;
+
+const TODOS_COLLECTION: &str = "todos";
+
+#[database("todo_service")]
+pub struct MyDatabase(Database);
+
+pub fn get_todos(conn: &MyDatabase) -> Result<Vec<Todo>, &str> {
+    let coll = conn.0.collection(TODOS_COLLECTION);
+    let cursor = coll.find(None, None).unwrap();
+    let mut todos: Vec<Todo> = Vec::new();
+
+    for result in cursor {
+        if let Ok(item) = result {
+            todos.push(Todo {
+                id: item.get_object_id("_id").unwrap().to_string(),
+                description: item.get("description").unwrap().to_string(),
+            });
+        }
+    }
+
+    Ok(todos)
+}
+
+pub fn get_todo(conn: &MyDatabase, id: String) -> Result<Todo, &str> {
+
+    // MongoDB IDs are ObjectIds not strings
+    let object_id = ObjectId::with_string(&id).unwrap();
```

Comment:
I was trying to keep the API implementation as non-database-specific as possible so that it would be a good general purpose reference for a web API. However, seeing as this is a MongoDB specific example, I can see how that would fine. Your call.